File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/vo/menu/ResourceListVO.java
package ink.ckx.mo.system.api.model.vo.menu;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.util.List;

@Schema(description = "资源列表VO")
@Data
public class ResourceListVO {

    @Schema(description = "选项的值")
    private Long value;

    @Schema(description = "选项的标签")
    private String label;

    @Schema(description = "子菜单")
    private List<ResourceListVO> children;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/vo/menu/ResourceListVO.kt
package ink.ckx.mo.admin.api.model.vo.menu

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "资源列表VO")
data class ResourceListVO(

    @Schema(description = "选项的值")
    var value: Any? = null,

    @Schema(description = "选项的标签")
    var label: String? = null,

    @Schema(description = "子菜单")
    var children: List<ResourceListVO>? = null,
)